File: two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  const hash = {};
  for (let i = 0; i < nums.length; i++) {
    const num = nums[i];
    if (hash[target - num] !== undefined) {
      return [i, hash[target - num]];
    }
    hash[num] = i;
  }
  return [];
}

File: two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map= new HashMap<Integer, Integer>(); 
        
        for(int i = 0; i < nums.length; i++) {
            int num = nums[i]; 
            
            if (map.get(target - num) != null) {
                return new int[] {i, map.get(target - num)};
            }
            
            map.put(num, i);
        }
        return new int[] {-1, -1};
    }
}